### PR TITLE
Add validation that the version has been increased compared to master

### PR DESCRIPTION
**Pull request by dekatzenel**
# Citrine Python PR

## Description 
Add automated validation that the version in setup.py has been bumped

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

### Adherence to team decisions
- [ ] I have added tests for 100% coverage
- [ ] I have written Numpy-style docstrings for every method and class.
- [x] I have communicated the downstream consequences of the PR to others.

### CODE REVIEW

**Comment by dekatzenel** [comment]
Success - the correct failure this time!

**Comment by dekatzenel** [comment]
Rebuild me!

**Comment by dekatzenel** [comment]
@maxhutch @mcgalcode In addition to general correctness, I would appreciate a sanity check/improvement recommendations on my bash script. If you do a review, please let me know if this is something you feel confident in doing

File: setup.py [hunk 1]
```diff
@@ -57,7 +57,7 @@ def run(self):
 
 
 setup(name='citrine',
-      version='0.9.1',
+      version='0.10.0',
```
Comment (maxhutch) [inline setup.py:60]: Isn't this a patch bump?

Comment (dekatzenel) [reply setup.py:60]: I'm including all the things I forgot

File: scripts/validate-version-bump.sh [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      exit 0
+    elif [ "${CURRENT_PATCH}" -lt "${MASTER_PATCH}" ]; then
+      echo "error - patch version decreased!"
+      exit 3
+    else
+      echo "error - version unchanged!"
+      exit 4
+    fi
+  fi
+fi
```
Comment (cyphus) [inline scripts/validate-version-bump.sh:59]: I gave this bash script a cursory glance and everything here is good shell scripting:
- `set -eu` 👍 I normally add in `-o pipefail` but there's no piping here so it's N/A
- quoted arguments to test builtin
- straightforward control flow

**Pull request merged**